operators/README.md

```diff
@@ -1,53 +1,91 @@
+
+All those resources are binded to the LabInstance life-cycle via the [OwnerRef property](https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/)
 
 ### Installation
-1. run `make install`. This will install the CRD LabInstance on your cluster.
-2. in _config/samples_ you have an example of LabInstance object. If you want to create it run `kubectl apply -f config/samples/instance_v1_labinstance.yaml`
-3. you can get the list of CRD installed on your cluster by running `kubectl get crd`.
-4. To get the list of LabInstance resources run `kubectl get labi`.
 
-If you want to delete the CRD run `make uninstall`.
+#### Pre-requirements
+
+The only LabOperator requirement is to have Kubevirt deployed.
+This can be done with the following commands, as reported by the official website:
+
+```bash
+# On other OS you might need to define it like
+export KUBEVIRT_VERSION="v0.34.0"
+
+# Deploy the KubeVirt operator
+kubectl create -f https://github.com/kubevirt/kubevirt/releases/download/${KUBEVIRT_VERSION}/kubevirt-operator.yaml
+# Only if HW Virtualization is not available
+kubectl create configmap kubevirt-config -n kubevirt --from-literal debug.useEmulation=true
+# Deploy Kubevirt
+kubectl create -f https://github.com/kubevirt/kubevirt/releases/download/${KUBEVIRT_VERSION}/kubevirt-cr.yaml
+```
+
+#### Deployment
+To deploy the LabOperator in your cluster, you have to do the following steps.
+
+First, set the desired values in `operators/labInstance-operator/k8s-manifest-example.env` .
+
+Then export the environment variables and generate the manifest from the template using:
+```
+export $(xargs < k8s-manifest-example.env)
+envsubst < k8s-manifest.yaml.tmpl > k8s-manifest.yaml
+```
+
+After the manifest have been correctly generated. You can deploy the labOperator using:
+
+```
+kubectl apply -f k8s-manifest.yaml
+```
+
+### Build from source
+
+LabOperator requires Golang 1.13 and make. To build the operator:
+
+```bash
+go build ./cmd/laboratory-operator/main.go
+```
+
+#### Testing
+
+After having installed Kubevirt in your testing cluster, you have to deploy the Custom Resource Definitions (CRDs) on the target cluster:
 
-### Controller logic
-The logic of the controller should be put under _controllers/labinstance_controller.go_, in the `Reconcile` method.
-When a LabInstance resource is created, the `Reconcile` method is triggered. 
-It is checked if a LabTemplate resource with the associated name exists, and in this case all the resources necessary to launch the Lab are created,
-in particular the `VirtualMachineInstance` CR managed by KubeVirt. 
+```bash
+make install
+make install-lab-template
+```
 
-### Run instructions
-To run the application
+Then, you can launch locally your operator:
 
-- **outside** a cluster: run 'make run'
-- **inside** a cluster: create a Pod with `container.image: crownlabs/laboratory-operator`
+```bash
+make run
```

Comment:
There is no longer a `make run` target in the makefile, as it would become meaningless when adding new operators.